src/types.js

```diff
@@ -324,3 +324,20 @@ export type PmConversationData = {|
   recipients: string,
   unread: number,
 |};
+
+export type SharedText = {|
+  type: 'text',
+  sharedText: string,
+|};
+
+export type SharedImage = {|
+  type: 'image',
+  sharedImageUrl: string,
+|};
+
+export type SharedFile = {|
+  type: 'file',
+  sharedFileUrl: string,
+|};
+
+export type SharedData = SharedText | SharedImage | SharedFile;
```

Comment:
This type should have jsdoc to explain what it's about. In particular that it represents a user action of sharing something from another app.

Reply:
Added a short jsdoc.